accounts/abi/bind/backends/blockchain.go

```diff
@@ -141,3 +144,197 @@ func (b *BlockchainContractCaller) getBlockAndState(num *big.Int) (*types.Block,
 	return block, state, err
 }
+
+type BlockchainContractTransactor struct {
+	bc     BlockChainForCaller
+	txPool *blockchain.TxPool
+}
+
+// This nil assignment ensures at compile time that BlockchainContractCaller implements bind.ContractCaller.
+var _ bind.ContractTransactor = (*BlockchainContractTransactor)(nil)
+
+func NewBlockchainContractTransactor(bc BlockChainForCaller, txPool *blockchain.TxPool) *BlockchainContractTransactor {
+	return &BlockchainContractTransactor{bc, txPool}
+}
+
+func (b *BlockchainContractTransactor) PendingCodeAt(ctx context.Context, account common.Address) ([]byte, error) {
+	// TODO-Klaytn this is not pending code but latest code
+	state, err := b.bc.State()
+	if err != nil {
+		return nil, err
+	}
+	return state.GetCode(account), nil
+}
+
+func (b *BlockchainContractTransactor) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	if b.txPool != nil {
+		return b.txPool.GetPendingNonce(account), nil
+	}
+	// TODO-Klaytn this is not pending nonce but latest nonce
+	state, err := b.bc.State()
+	if err != nil {
+		return 0, err
+	}
+	return state.GetNonce(account), nil
+}
+
+func (b *BlockchainContractTransactor) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
+	if b.bc.Config().IsMagmaForkEnabled(b.bc.CurrentHeader().Number) {
+		return new(big.Int).SetUint64(b.bc.Config().Governance.KIP71.UpperBoundBaseFee), nil
+	} else {
+		return new(big.Int).SetUint64(b.bc.Config().UnitPrice), nil
+	}
+}
+
+func (b *BlockchainContractTransactor) EstimateGas(ctx context.Context, call klaytn.CallMsg) (uint64, error) {
+	state, err := b.bc.State()
+	if err != nil {
+		return 0, err
+	}
+	balance := state.GetBalance(call.From) // from can't be nil
+
+	// Create a helper to check if a gas allowance results in an executable transaction
+	executable := func(gas uint64) (bool, *blockchain.ExecutionResult, error) {
+		call.Gas = gas
+
+		currentState, err := b.bc.State()
+		if err != nil {
+			return true, nil, nil
+		}
+		bc := NewBlockchainContractCaller(b.bc)
+		res, err := bc.callContract(call, b.bc.CurrentBlock(), currentState)
+		if err != nil {
+			if errors.Is(err, blockchain.ErrIntrinsicGas) {
+				return true, nil, nil // Special case, raise gas limit
+			}
+			return true, nil, err // Bail out
+		}
+		return res.Failed(), res, nil
+	}
+
+	estimated, err := blockchain.DoEstimateGas(ctx, call.Gas, 0, call.Value, call.GasPrice, balance, executable)
+	if err != nil {
+		return 0, err
+	} else {
+		return uint64(estimated), nil
+	}
+}
+
+func (b *BlockchainContractTransactor) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	if b.txPool == nil {
+		return errors.New("tx pool not configured")
+	}
+	return b.txPool.AddLocal(tx)
+}
+
+func (b *BlockchainContractTransactor) ChainID(ctx context.Context) (*big.Int, error) {
+	return b.bc.Config().ChainID, nil
+}
+
+type BlockchainContractFilterer struct {
+	bc BlockChainForCaller
+	es *filters.EventSystem
```

Comment:
`events` seems more intuitive rather than `es`

Reply:
Updated in c2b0f3d, thanks!